### PR TITLE
[v12] Install libbpf 1.0.1 in buildboxes

### PR DESCRIPTION
**Pull request by jakule**
Backport of #21596

### CODE REVIEW

**Comment by jakule** [comment]
@codingllama @atburke Friendly ping

File: build.assets/Dockerfile [hunk 1]
```diff
@@ -54,17 +54,33 @@ RUN git clone --depth=1 https://github.com/Yubico/libfido2.git -b 1.12.0 && \
...
+# Install required dependencies
 RUN apt-get update -y --fix-missing && \
     apt-get -q -y upgrade && \
     apt-get install -q -y --no-install-recommends \
         libelf-dev
 
 ARG LIBBPF_VERSION
-RUN mkdir -p /opt && cd /opt && curl -L https://github.com/gravitational/libbpf/archive/refs/tags/v${LIBBPF_VERSION}.tar.gz | tar xz && \
+RUN mkdir -p /opt && cd /opt &&  \
+    curl -L https://github.com/gravitational/libbpf/archive/refs/tags/v${LIBBPF_VERSION}.tar.gz | tar xz && \
```
Comment (codingllama) [inline build.assets/Dockerfile:65]:
```suggestion
    curl -fsSL https://github.com/gravitational/libbpf/archive/refs/tags/v${LIBBPF_VERSION}.tar.gz | tar xz && \
```

Context: https://github.com/gravitational/teleport/pull/22380

File: build.assets/Dockerfile [hunk 2]
```diff
@@ -54,17 +54,33 @@ RUN git clone --depth=1 https://github.com/Yubico/libfido2.git -b 1.12.0 && \
...
+FROM buildpack-deps:18.04 AS libbpf-1-0-1
+
+# Install required dependencies
+RUN apt-get update -y --fix-missing && \
+    apt-get -q -y upgrade && \
+    apt-get install -q -y --no-install-recommends \
+        libelf-dev
+
+RUN mkdir -p /opt && cd /opt && \
+    curl -L https://github.com/libbpf/libbpf/archive/refs/tags/v1.0.1.tar.gz | tar xz && \
```
Comment (codingllama) [inline build.assets/Dockerfile:80]:
```suggestion
    curl -fsSL https://github.com/libbpf/libbpf/archive/refs/tags/v1.0.1.tar.gz | tar xz && \
```

File: build.assets/Dockerfile-centos7 [hunk 3]
```diff
@@ -96,6 +90,27 @@ RUN mkdir -p /opt && cd /opt && \
...
+    yum install -y \
+        centos-release-scl \
+        devtoolset-11-gcc* \
+        devtoolset-11-make \
+        elfutils-libelf-devel-static \
+        scl-utils \
+    yum clean all
+
+RUN mkdir -p /opt && cd /opt && \
+    curl -L https://github.com/libbpf/libbpf/archive/refs/tags/v1.0.1.tar.gz | tar xz && \
```
Comment (codingllama) [inline build.assets/Dockerfile-centos7:110]:
```suggestion
    curl -fsSL https://github.com/libbpf/libbpf/archive/refs/tags/v1.0.1.tar.gz | tar xz && \
```

**Review by codingllama — APPROVED** [review]
My bad, I thought I approved this yesterday.

Added some `curl` suggestions, but you don't have to take them here in the backport - it's likely better to push from master again.

**Review by atburke — APPROVED** [review]

**Pull request merged**